openmdao/core/driver.py

```diff
@@ -474,7 +475,14 @@ def set_desvar(self, name, value):
         val : ndarray or float
             value to assign to the design variable.
         """
-        val = self.root.unknowns._dat[name].val
+        # support for uncertain samples
+        if name in self.root.unknowns._dat.keys():
+            val = self.root.unknowns._dat[name].val
+        elif re.findall("(.*)\[(.*)\]", name)[0][0] in self.root.unknowns._dat.keys():
+            vn =re.findall("(.*)\[(.*)\]", name)[0][0] 
+            nv = int(re.findall("(.*)\[(.*)\]", name)[0][1])
+            self.root.unknowns[vn][nv] = value
+            return
```

Comment:
Independent of the other comment, this `return` is too early, as it misses the code that applies `scaler` and `adder` to the design variable. The `scaler` and `adder` are arguments to `add_desvar` that allow the driver to see a transformed version of this variable for better numerical stability.